### ISSUE TITLE
Eslint import plugin not working when using pnpm

### ISSUE THREAD

**Issue by danialdezfouli** [microsoft/vscode-eslint#2070]
Type: <b>Bug</b>

There is no problem when using yarn or npm, but when use pnpm as a package manager vscode eslint extension doesn't work with eslint import package. Note that the eslint using working fine with no system errors in terminal and I'm using node v24 and already tried v22 using fnm. 

[error] An unexpected error occurred:
[error] Error: Cannot find native binding. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing both package-lock.json and node_modules directory.
Occurred while linting [...]/file.tsx:1
Rule: "import/namespace"
    at Object.<anonymous> ([PROJECT_PATH]/node_modules/.pnpm/unrs-resolver@1.11.1/node_modules/unrs-resolver/index.js:376:11)
    at Module._compile (node:internal/modules/cjs/loader:1738:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1904:10)
    at Module.load (node:internal/modules/cjs/loader:1472:32)
    at Module._load (node:internal/modules/cjs/loader:1289:12)
    at c._load (node:electron/js2c/node_init:2:18013)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:242:24)
    at Module.require (node:internal/modules/cjs/loader:1494:12)
    at require (node:internal/modules/helpers:135:16)
2025-08-29 13:53:46.574 [error] An unexpected error occurred:
2025-08-29 13:53:46.574 [error] Error: Cannot find native binding. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828). Please try `npm i` again after removing both package-lock.json and node_modules directory.

Extension version: 3.0.16
VS Code version: Code 1.103.2 (6f17636121051a53c88d3e605c491d22af2ba755, 2025-08-20T16:45:34.255Z)
OS version: Darwin arm64 25.0.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Pro (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: enabled_on<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|2, 3, 3|
|Memory (System)|16.00GB (0.09GB free)|
|Process Argv|--crash-reporter-id cb38fb92-7f77-44ce-afb2-99e284bfba77|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
binariesv615:30325510
nativeloc1:31344060
dwcopilot:31170013
dwoutputs:31242946
copilot_t_ci:31333650
e5gg6876:31282496
pythoneinst12:31285622
996jf627:31283433
pythonrdcb7:31342333
usemplatestapi:31297334
747dc170:31275177
6518g693:31334701
aj953862:31281341
9d2cg352:31346308
nesew2to5:31336538
agentclaude:31350858
nes-set-on:31351930
testaa123:31335226
onetestforazureexpcf:31335614
6abeh943:31336334
envsdeactivate2:31353495
0927b901:31350571
f76d9909:31348711
custommodel2t:31371781
45650338:31358607
0cj2b977:31352657
justeven_python:31371804
0574c672:31362109
cloudbuttont:31374192
retryenabled:31370050
3efgi100_wstrepl:31374188

```

</details>

**Comment by dbaeumer** [comment]
@danialdezfouli can you please provide me with a GitHub repository I can clone that demos this in a minimal setup. Errors like these most of the time depend on version, configs, ...

**Comment by matburnham** [comment]
Ah. Same/similar issue I've just raised at https://github.com/un-ts/eslint-plugin-import-x/issues/429 with a minimal repository at https://github.com/matburnham/broken-native-bindings

**Comment by anthonyjoeseph** [comment]
seems related to this issue with `eslint-import-resolver-typescript`: https://github.com/import-js/eslint-import-resolver-typescript/issues/431#issuecomment-2811859850

This worked for me, as a temporary fix:

```
pnpm i --save-dev @unrs/resolver-binding-wasm32-wasi
```

**Comment by dbaeumer** [comment]
Actually there is little from the ESLint extension side I can do about this. I will close the issue as upstream.

**Comment by danialdezfouli** [comment]
What about the fact that the same setup works well on WebStorm?

**Comment by dbaeumer** [comment]
Actually I cloned the workspace provided by @matburnham and it works for me.

<img width="1321" height="480" alt="Image" src="https://github.com/user-attachments/assets/99f4befb-807c-4563-bb5f-e8f7fc7fd7d3" />

**Comment by dbaeumer** [comment]
Is there anything I am missing?

**Comment by danialdezfouli** [comment]
you had to use pnpm, not npm

**Comment by dbaeumer** [comment]
I did

<img width="1533" height="590" alt="Image" src="https://github.com/user-attachments/assets/c8801064-6a0f-4d73-b207-dd291e758cd1" />

**Comment by matburnham** [comment]
Hmmm. Tested it midweek and it works for me now too. 🤷‍♂️ VSCode appears to have updated itself to 1.104.0 though. I don't know exactly what version it was on before but presumably whatever was previously released as it's auto-updated itself. Frustrating I don't have a before and after comparison. Were you on the 1.104.0 when you tried it @dbaeumer?

Had a bit more fun getting my non-minimum-test project playing nicely but that seems to be doing so too now.

**Comment by dbaeumer** [comment]
I tried it on latest insider.
